modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/components/field-sets/FieldSetList.js

```diff
@@ -57,11 +66,32 @@ export default function FieldSetList({searchTerm}) {
 	} = useFormState();
 
 	const {allowInvalidAvailableLocalesForProperty, fieldTypes} = useConfig();
-
 	const dispatch = useForm();
 
 	const filteredFieldsets = getFilteredFieldsets(fieldSets, searchTerm);
 
+	useEffect(() => {
+		const fieldsInuse = new Set();
+
+		new PagesVisitor(pages).mapFields((field) => {
+			if (field.type === 'fieldset') {
+				fieldsInuse.add(field.ddmStructureId);
+			}
+		});
+		setFieldSetsInUse(fieldsInuse);
+	}, [pages]);
+
+	const toggleFieldSet = (fieldSet) => {
+		setModalState(({isVisible}) => ({
+			fieldSet,
+			isVisible: !isVisible,
+		}));
+	};
+
+	const deleteFieldSet = useNewDeleteFieldSet();
+
+	const propagateFieldSet = usePropagateFieldSet();
```

Comment:
This name is very confusing, I didn't understand the idea of 'propagate' very well, what is the purpose of this?